src/ert/config/queue_config.py

```diff
@@ -111,6 +120,23 @@ def create_local_copy(self) -> QueueConfig:
         )
 
 
+def _check_for_overwritten_queue_system_options(
+    selected_queue_system: QueueSystem, queue_system_options: List[Tuple[str, str]]
+) -> None:
+    temp_dict: Dict[str, List[str]] = defaultdict(list)
+    for option_string in queue_system_options:
+        if isinstance(option_string, tuple):
+            option_name = option_string[0]
+            option_value = option_string[1]
+            temp_dict.setdefault(option_name, []).append(option_value)
+    for option_name, option_values in temp_dict.items():
```

Comment:
if we split the function in two, the first and second for loop. then we can make the first into a generator. because the actual behaviour of this function is done in the second part. the first seems to just "structurize" the data